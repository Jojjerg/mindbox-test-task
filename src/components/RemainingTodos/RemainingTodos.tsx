import { ITodo } from "../../@types/todo";
import React from "react";

const RemainingTodos = ({ todos }: {todos: ITodo[] }) => {
  const remaining = todos ? todos.filter((todo) => !todo.completed).length : '0';

  return (
    <div>
      <span data-testid="remaining-text">Осталось: {remaining}</span>
    </div>
  );
};

export default RemainingTodos;
