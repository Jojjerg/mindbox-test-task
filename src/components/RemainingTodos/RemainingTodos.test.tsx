import { render, screen } from '@testing-library/react';

import React from 'react';
import RemainingTodos from './RemainingTodos';
import { createRenderer } from 'react-test-renderer/shallow';

const renderer = createRenderer();

const todos = [
  {
    userId: 1,
    id: 1,
    title: "delectus aut autem",
    completed: false,
  },
  {
    userId: 1,
    id: 2,
    title: "quis ut nam facilis et officia qui",
    completed: true,
  },
];

const defaultComponent = <RemainingTodos todos={todos} />;

describe('<RemainingTodos />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(defaultComponent);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("remaining in component should be equal to 1", () => {
    const { getByTestId } = render(defaultComponent);
    const remainingText = getByTestId("remaining-text");
    
    expect(remainingText).toBeInTheDocument();
    expect(remainingText).toHaveTextContent("Осталось: 1")
  })
})