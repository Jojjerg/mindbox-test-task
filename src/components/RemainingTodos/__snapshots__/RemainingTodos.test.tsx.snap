// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemainingTodos /> should render and match the snapshot 1`] = `
<div>
  <span
    data-testid="remaining-text"
  >
    Осталось: 
    1
  </span>
</div>
`;
